### PR TITLE
docs(typescript): update example https server

### PR DESCRIPTION
**Pull request by Tony133**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Proposal:

- Updated the tutorial in the "example 2: server HTTPS" in the section on TypeScript

issue: #5387

### CODE REVIEW

File: docs/Reference/TypeScript.md [hunk 1]
```diff
@@ -862,9 +862,8 @@ a more detailed http server walkthrough.
    import path from 'path'
    import fastify from 'fastify'
    ```
-2. Follow the steps in this official [Node.js https server
-   guide](https://nodejs.org/en/knowledge/HTTP/servers/how-to-create-a-HTTPS-server/)
-   to create the `key.pem` and `cert.pem` files
+2. Follow the official documentation of [Node.js in the https section ](https://nodejs.org/api/https.html#httpscreateserveroptions-requestlistener)
```
Comment (climba03003) [inline docs/Reference/TypeScript.md:865]: It doesn't seems correct.

https://web.archive.org/web/20201217160232/https://nodejs.org/en/knowledge/HTTP/servers/how-to-create-a-HTTPS-server/

You should provide the `openssl` comment.

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: docs/Reference/TypeScript.md (hunk 1 above)
Comment (Tony133) [reply docs/Reference/TypeScript.md:865]: Okay then I'll add this link: 
https://web.archive.org/web/20201217160232/https://nodejs.org/en/knowledge/HTTP/servers/how-to-create-a-HTTPS-server/

 Is i leave the text as it was before

Comment (climba03003) [reply docs/Reference/TypeScript.md:865]: Just copy the commend in the link and write inside the doc.

Comment (Tony133) [reply docs/Reference/TypeScript.md]: Updated description see here: https://github.com/fastify/fastify/pull/5389/commits/4ec3b6c30637603b4fe2b5321170fe9f7070c21a

Comment (Tony133) [reply docs/Reference/TypeScript.md]: Lint Docs / Lint Markdown fails

Comment (Tony133) [reply docs/Reference/TypeScript.md]: now it is ok

**Review by climba03003 — APPROVED** [review]
Good, but better to copy the command inside and remove the link.

**Comment by Tony133** [comment]
> Good, but better to copy the commend inside and remove the link.

If I understand correctly, we leave the same text but without linking?  correct ?

**Review by metcoder95 — APPROVED** [review]
LGTM with @climba03003 comments

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by climba03003** [comment]
You still forgotten the part of `copy the commends inside the link in this doc` and doing just `remove the link`.

**Comment by Tony133** [comment]
> You still forgotten the part of `copy the commends inside the link to this doc` and doing just `remove the link`.

@climba03003  I set it up like this:

https://github.com/fastify/fastify/pull/5389/commits/6108f6407ba3b407d6a6afb1304d94c24ca76850dee

But, I still haven't figured out how you want to layout this part, can you send a change request? so I can see

**Comment by climba03003** [comment]
For what I want is the commend of generating `key.pem` and `cert.pem` exists in the document.

So, people don't need to click the link or rely on external source to figure out how to do.

**Comment by Tony133** [comment]
> Perché quello che voglio è l'elogio della generazione `key.pem`ed `cert.pem`esiste nel documento.
> 
> Gli `openssl`elogi sono esattamente all'interno del collegamento che fornisco.
> 
> Pertanto, le persone non hanno bisogno di fare clic sul collegamento o di affidarsi a fonti esterne per capire come farlo.

@climba03003  then we could write something like this:

2. Follow the steps in the official guide to the Node.js https server  
and create the `key.pem` and `cert.pem` files as follows:
```sh
openssl genrsa -out key.pem
openssl req -new -key key.pem -out csr.pem
openssl x509 -req -days 9999 -in csr.pem -signkey key.pem -out cert.pem
rm csr.pem
```


Do you like it like this ?

**Comment by climba03003** [comment]
Similar, but the wording need to change because there is no official documentation shown anymore.

cc @Fdawgs

**Comment by Tony133** [comment]
I have rephrased the description without reference to the official Node.js documentation, see commit here: https://github.com/fastify/fastify/pull/5389/commits/4d0d2b13d0539f8d6d7895d845d9732ae860fabb, please let me know if that will work for you 😉

Have a good week!

**Review by Fdawgs — CHANGES_REQUESTED** [review]
Will review tomorrow

File: docs/Reference/TypeScript.md [hunk 2]
```diff
@@ -862,9 +862,14 @@ a more detailed http server walkthrough.
    import path from 'path'
    import fastify from 'fastify'
    ```
-2. Follow the steps in this official [Node.js https server
-   guide](https://nodejs.org/en/knowledge/HTTP/servers/how-to-create-a-HTTPS-server/)
-   to create the `key.pem` and `cert.pem` files
+2. Follow the following steps before setting up the Fastify https server 
+for create the `key.pem` and `cert.pem` files, as follows:
```
Comment (Fdawgs) [inline docs/Reference/TypeScript.md:866]:
```suggestion
2. Perform the following steps before setting up a Fastify HTTPS server 
to create the `key.pem` and `cert.pem` files:
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**